integration-libs/cdp/root/components/cdp-my-account/cdp-my-account.component.ts

```diff
@@ -4,143 +4,75 @@
 
-  constructor(
-    protected routing: RoutingService,
-    protected orderHistoryFacade: OrderHistoryFacade,
-    protected translation: TranslationService,
-    protected replenishmentOrderHistoryFacade: ReplenishmentOrderHistoryFacade,
-    private userIdService: UserIdService,
-    private cdpOrderAdapter: cdpOrderAdapter,
-    protected occEndpointsService: OccEndpointsService,
-    protected datePipe: CxDatePipe,
-    @Optional() protected featureConfigService?: FeatureConfigService
-  ) {
-    super(
-      routing,
-      orderHistoryFacade,
-      translation,
-      replenishmentOrderHistoryFacade
-    );
-  }
-  result: finalOrder = { orders: [] };
-  totalPrice: number = 0;
-  totalItem: number[] = [];
-  orderDetail: Record<string, order> = {};
-  orderedItems: Record<string, number> = {};
-  i: number = 0;
-  output: result;
-  orderStatus: Record<string, Record<string, number>> = {};
-  orderImage: Record<string, product[]> = {};
-  userId: string;
-  tabTitleParam$ = new BehaviorSubject(0);
-  public loading$ = new BehaviorSubject<boolean>(true);
-  length: number = 0;
+  export class CdpMyAccountComponent implements OnInit {
+    constructor(
+      protected datePipe: CxDatePipe,
+      protected routing: RoutingService,
+      protected occEndpointsService: OccEndpointsService,
+      protected translation: TranslationService,
+      protected cdpOrderService: CdpOrderService,
+    ) {}
 
-  // orders$ = this.userIdService.takeUserId().pipe(switchMap((userId) => this.cdpOrderAdapter.getOrder(userId)));
+    orderValue: finalOrder = { orders: [] };
+    totalPrice: number = 0;
+    totalItem: number[] = [];
+    orderDetail: Record<string, order> = {};
+    orderStatus: Record<string, Record<string, number>> = {};
+    orderImage: Record<string, product[]> = {};
+    userId: string;
+    tabTitleParam$ = new BehaviorSubject(0);
+    public loading$ = new BehaviorSubject<boolean>(true);
+    sortType: string;
+    obser$: Observable<finalOrder>;
+    page_size: number = 3;
 
-  private P_SIZE = 3;
-  sortType: string;
-  hasPONumber: boolean | undefined;
-  // this.loading$.next(true);
-  orders$: Observable<OrderHistoryList | undefined> =
-    this.orderHistoryFacade.getOrderHistoryList(this.P_SIZE);
-  // this.loading$.next(false);
-  ngOnInit(): void {
-    this.getMyData();
-  }
+    ngOnInit(): void {
+      this.getMyData();
+    }
 
-  public getMyData(): void {
-    this.orders$.subscribe((res) => {
-      this.orders = res;
-      this.getOrderedItems(this.orders);
-      // this.length=Object.keys(this.orders).length;
-    });
-  }
+    public getMyData(): void {
+      this.obser$ = this.cdpOrderService.getOrder(this.page_size);
+      this.obser$.subscribe((res) => {
+        this.orderValue = res;
+        this.tabTitleParam$.next(res.orders.length);
+        this.getOrderedItems(this.orderValue);
+        console.log(this.orderValue);
 
-  public async getOrderedItems(orders: any): Promise<void> {
```

Comment:
would recommend renaming it to `getItemsInOrder(orders: Order)`.

Please use strict typing, avoid using `any` unless absolutely needed.